media_driver/agnostic/gen11/codec/kernel_free/media_srcs.cmake

```diff
@@ -18,6 +18,35 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+if (BUILD_KERNELS)
+
+    function(gen_cm_kernel_from_source name platform)
+
+        set(cm_krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/codec/kernel_free)
+        set(cm_krn ig${name}krn_g11)
+
+        # Generating source from the .krn file
+        add_custom_command(
+            OUTPUT ${cm_krn_dir}/${cm_krn}.c ${cm_krn_dir}/${cm_krn}.h
+            DEPENDS KernelBinToSource
```

Comment:
You need to list kernel source files in the DEPENDS otherwise you may not actually regenerate anything..